spec/requests/api/v1/csp_spec.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe 'API namespace minimal Content-Security-Policy' do
+  before do
+    # Replaces an already-routed-to API controller
```

Comment:
This is sort of awkward but I could not figure out how to set up an anonymous controller in a request spec. Tried this approach more or less - https://github.com/rspec/rspec-rails/issues/1596#issuecomment-834282306 - but kept getting routed to the not_found action ... I suspect there's a constraint or other route config somewhere that I'm not seeing and I need to do more on the route to set it up in the spec.

Reply:
Figured this out -- the `disable_clear_and_finalize` approach is usually good, but in our case the final "Catch all" route that routes to `application#raise_not_found` was taking precedence over what I was attempting to route to in the spec.

Added a commit which removes that setting and improves the setup here so that it actually is an anonymous controller (subclass of `Api::BaseController`) instead of replacing the existing one.